runtime/src/cost_model.rs

```diff
@@ -116,17 +100,11 @@ impl CostModel {
         tx_cost
     }
 
-    pub fn upsert_instruction_cost(
-        &mut self,
-        program_key: &Pubkey,
-        cost: u64,
-    ) -> Result<u64, &'static str> {
+    // update-or-insert should be infallible. Query result of upsert, which often
+    // requires additional calculation, should be lazy.
```

Comment:
nit: triple `///` For function documentation

This comment is confusing as it doesn't describe what the function is doing or provide any insight into how the function should be used. It seems to describe some design decision, but isn't very clear.

Reply:
It's more easily reviewed from https://github.com/solana-labs/solana/pull/23651/files?w=1

Reply:
right, good point. Will remove this comment next time near it